src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -694,11 +698,14 @@ handlePossibleRewind bRestore hsh = do
 
     newChildBlock bCurrent = do
         assign bsBlockHeight bRestore
-        SInt txid <- callDb "getting txid" $ \db ->
+        r <- callDb "getting txid" $ \db ->
           expectSingleRowCol msg =<< qry db
               "SELECT endingtxid FROM BlockHistory WHERE blockheight = ?;"
               [SInt (fromIntegral bCurrent)]
               [RInt]
+        txid <- case r of
+            SInt x -> return x
+            _ -> error "Chainweb.Pact.ChainwebPactDb.handlePossibleRewind.newChildBlock: failed to match SInt"
```

Comment:
These look like undetected partial patmatches? Yikes 😬

Reply:
I assume that the result being an `SInt` is a code invariant? This code comes from the db, so there is a serialization step and we'll have to reintroduce the invariant as a type at runtime. So the code is unsafe and we have to do something if our invariant doesn't hold, which would be a bug. I think using `error` is fine in that case.